Handle correction layer offsets

sssangha opened the pull request:
Leverage @mgovorcin sequential stitching implementation for SET and tropo correction layers.

Commit:
Handle correction layer offs

dbekaert commented:
per discussion; i beleive this is covering up an underlying problem in the SET code.

Commit:
Only correct for shift in SET, not tropo lyrs

Commit:
Fix bug regarding progbar initiation

Commit:
Do not attempt to stitch single frames

Commit:
Properly handle epochs for tropo in ariaTSsetup

Commit:
Address rioxarray update

sssangha committed:
Update requirements to remove python 3.8

Python 3.8 cannot handle the new `tuple` standard set for rolling versions of rioxarray

sssangha committed:
Remove python 3.11 check

Commit:
Adjust sequential stitcher to deal with single frames

dbekaert commented on tools/ARIAtools/tsSetup.py:
```diff
@@ -457,7 +461,7 @@ def main(inps=None):
         'outDir': inps.workdir,
         'outputFormat': inps.outputFormat,
-        'stitchMethodType': 'overlap',
+        'stitchMethodType': inps.stitchMethodType,
```
good catch

sssangha replied:
Yeah, but tbf I think that was intentional before... I.e. for ariaTSsetup the whole `stitchMethodType` argument had been commented out to enforce overlap stitching as 2stage unwrapping was only other choice and experimental.

dbekaert reviewed (approved):
LGTM So looks like we are not supporting 3.8 anymore with the recent changes.

sssangha commented:
> LGTM So looks like we are not supporting 3.8 anymore with the recent changes.

Yup, which reminds me to update the readme! In summary, the new version of rioxarray utilizes tuples in a way that is no longer compatible with [python3.8](https://docs.python.org/3.8/library/typing.html#type-aliases) vs python 3.9 and [above](https://docs.python.org/3.9/library/typing.html#type-aliases) as @cmarshak discussed with me..

Something to track in our other repos!

sssangha committed:
Supporting python 3.9+

Commit:
Cleanup to meet some pep8 standards

sssangha commented:
@mgovorcin I cleaned up the sequential stitcher a little bit, namely making it inline with pep8 standard of a max of 79 characters per line + adjusting the variables in the metadata layer stitcher I introduced

Commit:
Bug-fix for masking in sequential stitcher

Commit:
Search for existing keys

Commit:
turn off automatic plotting of stitcher
